src/CMakeLists.txt

```diff
@@ -12,4 +12,29 @@ project(vocoder VERSION 0.0)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-add_executable(vocoder src/main.cpp)
\ No newline at end of file
+include(FetchContent)
+FetchContent_Declare(
+	googletest
+	URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
+	)

```

Comment:
This is the recommended approach according to:
https://google.github.io/googletest/quickstart-cmake.html

And this does point to a particular version of google test, which they recommend updating. Not sure how we would get around this.

Reply:
Do we want to chase the bleeding edge of gtest, or should we pin to whatever version is out now?

Google considers it best practice to be on HEAD, but it would be simpler for us just to pin. I think for this project chasing HEAD might be the better idea...

Reply:
Is there a way to do this automatically, rather than pointing to a specific version? surprised you can't just do like URL ......googletest/HEAD.zip or some equivalent

Reply:
could try:

https://github.com/google/googletest/archive/refs/heads/main.zip

Reply:
Nope :(
![image](https://user-images.githubusercontent.com/46714617/219046187-742eab09-ab98-4212-93d1-ff95bb5ac2a4.png)

Reply:
ah well, good enough as is then